tirtha_bk/tirtha/views.py

```diff
@@ -163,39 +186,42 @@ def index(request, vid=None, runid=None):
             if not contrib.banned and contrib.active:
                 context.update({"signin_class": ""})
 
+    #block for profile image url
+    try:
+        context["profile_image_url"] = token.get('userinfo').get('picture') #adding the currently signed in profile image to the context
```

Comment:
Does this have a chance of failing? If not, you can remove the try-except. In any case, the exception is being ignored.

Reply:
It needs to stay, as the function "index" will run and it will render the html, by default no user will be signed in hence it should not update the "context" field, so it is in the code to prevent the internal error!

Reply:
Then, this can be in the `if` block. `token` is either `None` or a valid, `oauth`-verified dict. The `if` block checks for the `None` case.

If `token['userinfo']['picture']` can be `None`, then another if condition can be added to handle that. `try-except` is not applicable here.